backend/lib/endpoints/feedback.js

```diff
@@ -0,0 +1,36 @@
+const { createFeedbackSchema } = require("./schema/feedback");
+
+/*
+ * /api/feedback
+ */
+async function routes(app) {
+  const Feedback = app.mongo.model("Feedback");
+
+  app.post("/", { schema: createFeedbackSchema }, async (req, reply) => {
+    const { userId } = req.body;
+
+    if (userId) {
+      const [err, userFeedback] = await app.to(Feedback.find({ userId }));
+      if (userFeedback[0] || err) {
+        throw app.httpErrors.internalServerError();
+      }
+    }
+
+    const [err] = await app.to(
+      new Feedback({
+        ...req.body,
+        ipAddress: req.ip,
+      }).save(),
+    );
+
+    if (err) throw app.httpErrors.internalServerError();
+
+    setImmediate(() => {
+      reply.code(201).send({ success: true });
```

Comment:
why do you need this `setImmediate`? I would think you could directly `return reply` or maybe because that was not recommended by fastify docs for async/await, just do

```
reply.code(201);
return { success: true };
```

Also, I think usually the newly created item is returned, although for a feedback this is probably not used on the frontend...
but yeah I have to admit I don't really like that this feels like a workaround if we just want to return a different status code, maybe there's a better way, not that important though

Reply:
So the `setImmediate` was something taken directly from the docs with regard to `return reply`. But your suggestion is also pretty nice. Maybe we can find something nicer in the future.